app/src/main/java/com/android/sooz/chatapp/StatusActivity.java

```diff
@@ -0,0 +1,136 @@
+package com.android.sooz.chatapp;
+
+import android.content.Intent;
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.widget.EditText;
+
+import com.google.firebase.database.DataSnapshot;
+import com.google.firebase.database.DatabaseError;
+import com.google.firebase.database.DatabaseReference;
+import com.google.firebase.database.FirebaseDatabase;
+import com.google.firebase.database.ValueEventListener;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import butterknife.BindView;
+import butterknife.ButterKnife;
+import butterknife.OnClick;
+
+public class StatusActivity extends AppCompatActivity {
+
+    @BindView(R.id.setStatus)
+    public EditText mEditText;
+
+    @BindView(R.id.list)
+    public RecyclerView list;
+
+    private LinearLayoutManager linearLayoutManager;
+    private StatusAdapter statusAdapter;
+
+    //bring in user data from FirebaseDB
+    FirebaseDatabase mDatabase;
+    DatabaseReference mUsers;
+
+    //track the username of each user
+    private String mUsername;
+
+    List<ChatStatus> allStatuses;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_status);
+
+        ButterKnife.bind(this);
+
+        mDatabase = FirebaseDatabase.getInstance();
+        mUsers  = mDatabase.getReference("users");
+
+        attachListeners();
+        initializeUsername();
+
+        allStatuses = new ArrayList<>();
+        linearLayoutManager = new LinearLayoutManager(this);
+        statusAdapter = new StatusAdapter(allStatuses);
+
+        list.setLayoutManager(linearLayoutManager);
+        list.setAdapter(statusAdapter);
+    }
+
+    private void initializeUsername(){
+        Intent data = getIntent();
+        mUsername = data.getStringExtra("username");
```

Comment:
Rather than hardcoding "username", this should probably be defined as a constant.

Reply:
made this change, but please note, this was as done by Steve in class for getting username from the data.